File: programmers/Level01/Programmers_12943.java
package programmers;

public class Programmers_12943 {
    public int solution (int num) {
        int answer = 0;
        long l_num = num;

        while(l_num != 1 && answer < 500){
            if(l_num % 2 == 0){
                l_num /= 2;
            } else {
                l_num = 3 * l_num + 1;
            }
            answer++;
        }

        return answer == 500 ? -1 : answer;
    }
}

File: programmers/Swift/Level01/Programmers_12943.swift
final class Programmers_12943 {
    func solution(_ num:Int) -> Int {
        var n = num
        for i in 0...500 {
            if n == 1 {
                return i
            }
            n = n % 2 == 0 ? n / 2 : n * 3 + 1
        }
        return -1
    }
}